src/index.ts

```diff
@@ -331,45 +302,40 @@ task("starknet-deploy", "Deploys Starknet contracts which have been compiled.")
             const files = paths.filter(isStarknetCompilationArtifact);
             for(const file of files){
                 console.log("Deploying", file);
-                const executed = await hre.starknetWrapper.runCommand(
-                    "starknet",
-                    [
-                        "deploy",
-                        "--contract", file,
-                        "--gateway_url", adaptUrl(gatewayUrl),
-                        ...inputs
-                    ],
-                    [artifactsPath]
-                );
+                const executed = await hre.starknetWrapper.deploy({
+                    contract: file,
+                    gatewayUrl,
+                    inputs: args.inputs ? args.inputs.split(/\s+/) : undefined,
+                });
                 if(args.wait){
                     const execResult = processExecuted(executed);
                     if(execResult == 0){
                         txHashes.push(extractTxHash(executed.stdout.toString()));
                     }
                     statusCode += execResult;
                 } 
-                else
+                else {
                     statusCode += processExecuted(executed);
+                }
             }
         }
 
-        if(args.wait){      //  If the "wait" flag was passed as an argument, check the previously stored transaction hashes for their statuses
+        if (args.wait){ // If the "wait" flag was passed as an argument, check the previously stored transaction hashes for their statuses
             console.log("Checking deployment transactions...");
-            const promises = txHashes.map( hash => new Promise<void>((resolve, reject) => {iterativelyCheckStatus(
+            const promises = txHashes.map( hash => new Promise<void>((resolve, reject) => iterativelyCheckStatus(
                 hash, 
                 hre.starknetWrapper, 
                 gatewayUrl, 
                 gatewayUrl, 
                 () => {
-                    console.log("Deployment transaction " + hash + " status is now PENDING");
+                    console.log("Deployment transaction " + hash + " status is now at least PENDING");
```

Comment:
Should change to "at least ACCEPTED_ON_L2", right? That's what we are waiting for during the iterative check